analyses/tp53_nf1_score/00-tp53-nf1-alterations.R

```diff
@@ -0,0 +1,103 @@
+# Author: Krutika Gaonkar
+#
+# Read in concensus snv calls to gather alterations in TP53 and NF1
+# to evaluate classifier
+# @params snvConcensus multi-caller concensus snv calls
+# @params clincalFile clinical file: pbta-histologies.tsv
+# @params outputFolder output folder for alteration file
+# @params gencode cds bed file from gencode
+
+suppressPackageStartupMessages(library("optparse"))
+suppressPackageStartupMessages(library("tidyverse"))
+suppressPackageStartupMessages(library("readr"))
+suppressPackageStartupMessages(library("GenomicRanges"))
+
+#### Source functions ----------------------------------------------------------
+# We can use functions from the `snv-callers` module of the OpenPBTA project
+# TODO: if a common util folder is established, use that instead
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+source(file.path(root_dir, "analyses", "snv-callers", "util",
+                 "tmb_functions.R"))
+
+#### Parse command line options ------------------------------------------------
+
+option_list <- list(
+  make_option(c("-s", "--snvConsensus"),type="character",
+              help="Consensus snv calls (.tsv) "),
+  make_option(c("-c","--clinicalFile"),type="character",
+              help="clinical file for all samples (.tsv)"),
+  make_option(c("-o","--outputFolder"),type="character",
+              help="output folder for results "),
+  make_option(c("-g","--gencode"),type="character",
+              help="cds gencode bed file")
+)
+
+opt <- parse_args(OptionParser(option_list=option_list))
+snvConsensusFile <- opt$snvConsensus
+clinicalFile <- opt$clinicalFile
+outputFolder <- opt$outputFolder
+gencodeBed <- opt$gencode
+
+#### Generate files with TP53, NF1 mutations -----------------------------------
+
+# read in consensus SNV files
+consensus_snv <- read_tsv(snvConsensusFile)
```

Comment:
If these are the only columns you are using, this should speed up this step a bit, but mostly this will be sped up by using `data.table::fread` instead of `readr::read_tsv`

```suggestion
consensus_snv <-data.table::fread(snvConsensusFile, select = c("Chromosome", "Start_Position", "End_Position", "Strand", "Variant_Classification", "Tumor_Sample_Barcode", "Hugo_Symbol"), 
data.table = FALSE)
```

Reply:
`data.table::fread` is faster for big files.